api/v1/serviceinstance_types.go

```diff
@@ -105,16 +110,22 @@ type ServiceInstanceStatus struct {
 
 	// Indicates whether instance is ready for usage
 	Ready metav1.ConditionStatus `json:"ready,omitempty"`
+
+	// Indicates whether instance is shared
+	// +optional
+	Shared metav1.ConditionStatus `json:"shared,omitempty"`
```

Comment:
I need it because this is the source of truth 
We have shared in spec - which is the desired 
And we have shared in status - which is the current state
This was discussed in the design review

Reply:
i'm not sure we are talking about the same thing... 
what is the difference between conditions.shared.status and status.shared? is there a situation when conditions.shared.status=true and status.shared=false or vise versa?

Reply:
If the instance is shared, then the user request to unshare it, then the unshare failed

the condition will not be aligned

For example here

<img width="976" alt="image" src="https://user-images.githubusercontent.com/108805932/232714004-9471e7af-8057-4271-aa52-6bf562400e7d.png">